website/docs/04_design/package-config.md

```diff
@@ -0,0 +1,160 @@
+# Package Configuration
+
+:::info Disclaimer
+This is a design document.
+Some or all of the functionality described in the following may not be implemented in a released version of Glasskube.
+:::
+
+## Probelm analysis
+
+At Glasskube, what we mean when we talk about package configuration, is a controlled alteration of the resources that are part of a package depending on values specified by the user.
+This can be thought of as three sub-problems:
+
+### What values are available for a given package
+
+The author of a package may include a declaration of the possible configuration items in the package manifest, where each package may have multiple **value specifications**.
+A value specification contains information to help clients display an appropriate form item for data entry and constraints to be validated by the client as well as the package operator.
+
+### How does a value affect the deployed package resources
+
+Additionally, a value specification describes some alterations to the deployed package.
+We call those alterations **targets**.
+Each value specification may contain a number of targets.
+A target can be one of two things:
+
+1. A change to the default values of a helm release contained in the package
+2. A JSON patch that should be applied to a resource contained in a `manifests` entry of the package
+
+### How does a value affect the deployed package resources
+
+A user, when installing a package using Glasskube, may declare a **value** for each value specification of that package.
+A value can be either a **literal value** or a **reference value**.
+Literal values represent a simple value (e.g. a string entered via a text field).
+Reference values represent references to values in other resources in the same Kubernetes cluster.
+Such references can be secrets, configmaps or other packages.
+
+## Design proposal
+
+The `PackageManifest` has a property `Values` of type `map[string]ValueSpecification`.
+The key in this map is referred to as that values **name**
+`ValueSpecification` is a struct with the following properties:
+
+- **`Type`** (`string` enum):
+  Every value must have a type, so that we know what kind of input field to show for this value.
+  Initially, this can be one of `check`, `options`, `text`, `number` but it is possible to add more types in future releases.
```

Comment:
> I would call it boolean instead of check – that sounds too much like a UI element to me and I would clearly separate the type here from how it might be displayed by a client.

I think that's a good idea. Let's do that.

> We will also need to define null-values for each type, e.g. if a boolean value is not specified by the user and there is no DefaultValue, is it false or empty/undefined/null?

A non-specified value will not be applied to it's targets. The package author must ensure that there is a default value in the place where it would be applied, if a missing value should be supported.

> Also, will number also include floating point numbers?

To start with, I would say no. We could add a `hint` for this kind of thing in the future though.